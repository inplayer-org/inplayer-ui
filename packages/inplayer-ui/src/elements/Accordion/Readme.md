Accordion or expansion panel is a component that generates Accordion component with more accordion panels. It takes `panels` - array of objects and each object should contain:
<br/>- `label` (title)
<br/>- `icon` (icon displayed next to the title)
<br/> - `renderContent` (action used to render the content)
<br/> - `iconTooltip` (optional parameter - a tooltip to display on hovering the icon, see `Tooltip`'s prop list) - default `null`
<br/> - `disabled` (optional paremeter - a boolean to disable clicking or opening the panel) - default `false`
<br/>
<br/>Other props that the accordion accepts:
<br/> - `isExtendable` (display arrow and allow the accordion to extend)
<br/> - `width` (optional parameter - sets the width for the Accordion wrapper) - default 100%
<br/> - `extendWidth` (optional parameter - defines the width the accordion should extend) - default 20%
<br/> - `contentHeight` - that determines the height of the wrapper.
<br/> - `onActivePanelChange` - function to be executed when the accordion tab changes

```jsx static
const panels = [
  {
    label: 'Accordion1',
    icon: 'info-circle',
    renderContent: ({ closePanel }) => {
      return <h1>Content for the accordion1</h1>;
    },
  },
  {
    label: 'Accordion2',
    icon: 'info-circle',
    renderContent: ({ closePanel }) => {
      return <h5>Content for the accordionh2</h5>;
    },
  },
];
<Accordion panels={panels} contentHeight="46vh" />;
```

### Demo

```jsx
const panels = [
  {
    label: 'Accordion1',
    icon: 'info-circle',
    renderContent: ({ closePanel }) => {
      return (
        <div>
          <h1>Long Content for the accordion1</h1>
          <h1>Long Content for the accordion1</h1>
          <h1>Long Content for the accordion1</h1>
          <h1>Long Content for the accordion1</h1>
          <h1>Long Content for the accordion1</h1>
          <h1>Long Content for the accordion1</h1>
        </div>
      );
    },
  },
  {
    label: 'Accordion2',
    icon: 'info-circle',
    iconTooltip: {
      content: 'This is a disabled panel with optional icon with a tooltip',
      placement: 'left',
    },
    renderContent: ({ closePanel }) => {
      return <h5>Content for the accordionh2</h5>;
    },
    disabled: true,
  },
  {
    label: 'Accordion3',
    renderContent: () => {
      return <h5>Content for the accordionh3</h5>;
    },
  },
];

<Accordion
  onActivePanelChange={index => console.log(index)}
  panels={panels}
  contentHeight="46vh"
  isExtendable
  width="80%"
  extendWidth="20%"
/>;
```
