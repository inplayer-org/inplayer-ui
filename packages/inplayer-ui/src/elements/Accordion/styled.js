import styled, { css } from 'styled-components';
import colors from 'config/colors';

export const AccordionWrapper = styled.div`
  ${({ width }) => width && `width: ${width}`};
  margin-left: 0%;
  position: relative;
  background: ${colors.white};
  border-left: 1px solid ${colors.gray};
  box-sizing: border-box;
  border-top: 1px solid ${colors.gray};
  ${({ contentHeight }) => contentHeight && `height: calc(${contentHeight} - 120px)`};
  overflow-y: hidden;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  transition: width 0.3s ease;
  ${({ open }) =>
    open &&
    css`
      ${({ width, extendWidth }) => width && `width: calc(${width} + ${extendWidth})`};
    `}
`;
