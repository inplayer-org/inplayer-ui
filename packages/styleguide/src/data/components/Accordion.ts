import { ComponentData } from "../types";

const usage = `
  const panels = [
    {
      label: 'Accordion1',
      icon: 'info-circle',
      renderContent: ({ closePanel }) => {
        return <h1>Content for the accordion1</h1>;
      },
    },
    {
      label: 'Accordion2',
      icon: 'info-circle',
      renderContent: ({ closePanel }) => {
        return <h5>Content for the accordionh2</h5>;
      },
    },
  ];
  <Accordion panels={panels} contentHeight="46vh" />;
`;

const demoCode = `
Page = () => {
  const panels = [
    {
      label: 'Accordion1',
      icon: 'info-circle',
      renderContent: ({ closePanel }) => {
        return (
          <div>
            <h1>Long Content for the accordion1</h1>
            <h1>Long Content for the accordion1</h1>
            <h1>Long Content for the accordion1</h1>
            <h1>Long Content for the accordion1</h1>
            <h1>Long Content for the accordion1</h1>
            <h1>Long Content for the accordion1</h1>
          </div>
        );
      },
    },
    {
      label: 'Accordion2',
      icon: 'info-circle',
      iconTooltip: {
        content: 'This is a disabled panel with optional icon with a tooltip',
        placement: 'left',
      },
      renderContent: ({ closePanel }) => {
        return <h5>Content for the accordionh2</h5>;
      },
      disabled: true,
    },
    {
      label: 'Accordion3',
      renderContent: () => {
        return <h5>Content for the accordionh3</h5>;
      },
    },
  ];
  
  return <Accordion
    onActivePanelChange={index => console.log(index)}
    panels={panels}
    contentHeight="46vh"
    isExtendable
    width="80%"
    extendWidth="20%"
  />;
}
`;

const Accordion: ComponentData = {
  description: `
  Accordion or expansion panel is a component that generates Accordion component with more accordion panels. It takes panels - array of objects and each object should contain: 
  - label (title) 
  - icon (icon displayed next to the title) 
  - renderContent (action used to render the content) 
  - iconTooltip (optional parameter - a tooltip to display on hovering the icon, see Tooltip's prop list) - default null 
  - disabled (optional paremeter - a boolean to disable clicking or opening the panel) - default false 
  
  Other props that the accordion accepts: 
  - isExtendable (display arrow and allow the accordion to extend) 
  - width (optional parameter - sets the width for the Accordion wrapper) - default 100% 
  - extendWidth (optional parameter - defines the width the accordion should extend) - default 20% 
  - contentHeight - that determines the height of the wrapper. 
  - onActivePanelChange - function to be executed when the accordion tab changes
  
      `,
  path: 'src/elements/Accordion/index.js',
  usage,
  examples: [
    {
      title: 'Demo',
      code: demoCode,
    }
  ],
  propsAndMethods: [
    { name: 'contentHeight', type: 'string', default: 'Required' },
    { name: 'panels', type: 'Array<Panel>', default: 'Required' },
    { name: 'extendWidth', type: 'string', default: '20%' },
    { name: 'isExtendable', type: 'boolean', default: 'false' },
    { name: 'onActivePanelChange', type: 'function', default: 'null' },
    { name: 'width', type: 'string', default: '100%'},
  ],
}

export default Accordion;