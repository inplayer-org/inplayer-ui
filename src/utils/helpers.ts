import moment from 'moment';

export const getMonthOptions = (): Array<any> =>
  moment.months().map((label, value) => ({ value, displayName: label }));

export const getYearOptions = () => {
  const yearOptions: Array<any> = [];
  for (let i = moment().year() - 100; i <= moment().year(); i += 1) {
    yearOptions.push({ value: i.toString(), displayName: i.toString() });
  }

  return yearOptions;
};
