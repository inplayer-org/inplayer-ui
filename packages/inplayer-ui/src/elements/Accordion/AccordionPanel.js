// @flow
import React from 'react';
import styled, { css } from 'styled-components';
import colors from 'config/colors';
import { ifProp, ifNotProp } from 'styled-tools';
import { uiColors, fontWeights } from 'utils';

// Components
import Icon from 'elements/Icon';
import Typography from 'elements/Typography';
import Tooltip from 'components/Tooltip';
import type { Props as TooltipProps } from 'components/Tooltip';
import InPlayerIcon from 'elements/InPlayerIcon';

const AccordionPanelContainer = styled.div`
  width: 100%;
  height: 3.5rem;
  box-sizing: border-box;
  overflow: hidden;
  z-index: 10;
  background: ${colors.white};
  transition: ease 500ms height;
  ${ifProp(
    'isActive',
    css`
      position: absolute;
      ${({ contentHeight }) => contentHeight && `height: calc(${contentHeight} - 11rem)`};
      top: 0;
      z-index: 30;
      border-bottom: none;
      overflow: auto;
      margin-top: 3.5rem;
      display: block;
    `,
    css`
      height: 0;
    `
  )};
`;

const AccordionPanelHeader = styled.header`
  margin: 0;
  padding: 1rem 3%;
  width: 100%;
  box-sizing: border-box;
  background: ${ifProp('disabled', colors.lightGray, colors.white)};
  cursor: ${({ disabled }) => (disabled ? 'default' : 'pointer')};
  border-bottom: 1px solid ${colors.lightGray};
  display: flex;
  justify-content: space-between;
  align-items: center;
  ${ifProp(
    'isActive',
    css`
      position: absolute;
      top: 0;
    `
  )};
`;

const AccordionTitle = styled(Typography)`
  font-weight: ${fontWeights('thin')};
  color: ${ifProp('disabled', uiColors('text.disabled'), uiColors('text.main'))};
  margin: 0;
  display: inline;
`;

const AccordionIcon = styled(Icon)`
  color: ${uiColors('primary.main')};
  ${({ pointer }) => (pointer ? 'cursor: pointer;' : '')}
`;

const AccordionPanelDetails = styled.div`
  padding: 0;
  height: 100%;
`;

const AccordionIconHolder = styled.div`
  ${ifNotProp(
    'isAccordionDisabled',
    css`
      display: flex;
      width: 3rem;
      justify-content: space-between;
    `
  )};
`;

type Props = {
  label: string,
  isActive: boolean,
  isOtherPanelActive: boolean,
  icon?: string,
  iconTooltip?: TooltipProps,
  contentHeight: string,
  togglePanel: (panel: number) => (e?: SyntheticEvent<*>) => void,
  renderContent: (actions: { closePanel: (e?: SyntheticEvent<any>) => void }) => any,
  closePanel: (e?: SyntheticEvent<*>) => void,
  disabled: boolean,
};

const AccordionPanel = ({
  label,
  isActive,
  isOtherPanelActive,
  icon,
  iconTooltip,
  renderContent,
  contentHeight,
  disabled,
  togglePanel,
  closePanel,
}: Props) => (
  <>
    <AccordionPanelHeader
      disabled={disabled}
      onClick={!disabled ? togglePanel : null}
      isActive={isActive}
    >
      <AccordionTitle variant="h6" isActive={isActive} disabled={disabled}>
        {label}
      </AccordionTitle>
      <AccordionIconHolder isAccordionDisabled={disabled}>
        {!isOtherPanelActive &&
          (iconTooltip ? (
            <Tooltip {...iconTooltip}>
              <AccordionIcon name={icon} pointer />
            </Tooltip>
          ) : (
            <AccordionIcon name={icon} />
          ))}
        {!disabled && <InPlayerIcon name={isActive ? 'angleUp' : 'angleDown'} />}
      </AccordionIconHolder>
    </AccordionPanelHeader>
    <AccordionPanelContainer isActive={isActive} contentHeight={contentHeight}>
      <AccordionPanelDetails>{isActive && renderContent({ closePanel })}</AccordionPanelDetails>
    </AccordionPanelContainer>
  </>
);

AccordionPanel.defaultProps = {
  icon: '',
  iconTooltip: null,
};

export default AccordionPanel;
