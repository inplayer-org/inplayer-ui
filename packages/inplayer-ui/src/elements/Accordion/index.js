// @flow
import React, { Component } from 'react';

// Components
import type { Props as TooltipProps } from 'components/Tooltip';
import { AccordionWrapper } from './styled';
import AccordionPanel from './AccordionPanel';
import Arrow from '../../components/NavBar/Arrow';

// Types
type Panel = {
  label: string,
  icon?: string,
  iconTooltip?: TooltipProps,
  renderContent: () => any,
  disabled: boolean,
};

type Props = {
  panels: Array<Panel>,
  contentHeight: string,
  width?: string,
  extendWidth?: string,
  isExtendable?: boolean,
  onActivePanelChange?: (index: number) => void,
};

type State = { activePanel: number, open: boolean };

class Accordion extends Component<Props, State> {
  state = { activePanel: -1, open: false };

  openPanel = (panelIndex: number) => {
    const { activePanel } = this.state;
    const { onActivePanelChange } = this.props;

    if (panelIndex !== activePanel) {
      this.setState({ activePanel: panelIndex });
      if (onActivePanelChange) onActivePanelChange(panelIndex);
    }
  };

  closePanel = (e?: SyntheticEvent<*>) => {
    if (e) e.stopPropagation();
    this.setState({ activePanel: -1 });
  };

  togglePanel = (panelIndex: number) => (e: SyntheticEvent<*>) => {
    e.stopPropagation();
    const { activePanel } = this.state;
    if (activePanel !== -1) {
      this.closePanel();
    } else {
      this.openPanel(panelIndex);
    }
  };

  toggleOpen = () => {
    const { isExtendable } = this.props;
    if (isExtendable) this.setState(({ open }) => ({ open: !open }));
  };

  render() {
    const { panels, contentHeight, width, extendWidth, isExtendable } = this.props;

    const { state } = this;
    const { open, activePanel } = state;

    return (
      <AccordionWrapper
        width={width}
        extendWidth={extendWidth}
        open={open}
        contentHeight={contentHeight}
      >
        <div>
          {panels.map((panel, index) => {
            const { icon, iconTooltip, label, renderContent, disabled } = panel;
            const isActive = activePanel === index;
            const isOtherPanelActive = !isActive && activePanel !== -1;

            return (
              <div key={index}>
                <AccordionPanel
                  key={index}
                  isActive={isActive}
                  icon={icon}
                  iconTooltip={iconTooltip}
                  label={label}
                  renderContent={renderContent}
                  closePanel={this.closePanel}
                  isOtherPanelActive={isOtherPanelActive}
                  contentHeight={contentHeight}
                  disabled={disabled}
                  togglePanel={this.togglePanel(index)}
                />
              </div>
            );
          })}
        </div>
        {isExtendable && activePanel === -1 && (
          <Arrow onClick={this.toggleOpen} open={open} section="accordion" />
        )}
      </AccordionWrapper>
    );
  }
}

Accordion.defaultProps = {
  width: '100%',
  extendWidth: '20%',
  isExtendable: false,
  onActivePanelChange: null,
};

/** @component */
export default Accordion;
