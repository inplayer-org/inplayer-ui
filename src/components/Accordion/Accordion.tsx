import React, { useState, SyntheticEvent, ReactElement } from 'react';

// Components
import Arrow from 'components/NavBar/Arrow';
import { TooltipProps } from 'components/Tooltip/Tooltip';
import { AccordionWrapper } from './styled';
import AccordionPanel from './AccordionPanel';

// Types
type Panel = {
  label: string;
  icon?: ReactElement;
  iconTooltip?: TooltipProps;
  renderContent: () => any;
  disabled?: boolean;
};

type Props = {
  /**
   * Array of objects
   */
  panels: Array<Panel>;
  /**
   * Determines the height of the wrapper
   */
  contentHeight: string;
  /**
   * Sets the width for the Accordion wrapper
   */
  width?: string;
  /**
   * Defines the width the accordion should extend
   */
  extendWidth?: string;
  /**
   * Display arrow and allow the accordion to extend
   */
  isExtendable?: boolean;
  /**
   * Function to be executed when the accordion tab changes
   */
  onActivePanelChange?: (index: number) => void;
};

export const Accordion = ({
  panels,
  contentHeight = '20%',
  width = '100%',
  extendWidth = '20%',
  isExtendable = false,
  onActivePanelChange,
}: Props) => {
  const [activePanel, setActivePanel] = useState(-1);
  const [open, setOpen] = useState(false);

  const openPanel = (panelIndex: number) => {
    if (panelIndex !== activePanel) {
      setActivePanel(panelIndex);
      if (onActivePanelChange) onActivePanelChange(panelIndex);
    }
  };

  const closePanel = (e?: SyntheticEvent) => {
    if (e) e.stopPropagation();
    setActivePanel(-1);
  };

  const togglePanel = (panelIndex: number) => (e: any) => {
    e.stopPropagation();
    if (activePanel !== -1) {
      closePanel();
    } else {
      openPanel(panelIndex);
    }
  };

  const toggleOpen = () => {
    if (isExtendable) setOpen(!open);
  };

  return (
    <AccordionWrapper
      width={width}
      extendWidth={extendWidth}
      open={open}
      contentHeight={contentHeight}
    >
      <div>
        {panels.map((panel, index) => {
          const { icon, iconTooltip, label, renderContent, disabled = false } = panel;
          const isActive = activePanel === index;
          const isOtherPanelActive = !isActive && activePanel !== -1;

          return (
            <div key={index}>
              <AccordionPanel
                key={index}
                isActive={isActive}
                icon={icon}
                iconTooltip={iconTooltip}
                label={label}
                renderContent={renderContent}
                closePanel={closePanel}
                isOtherPanelActive={isOtherPanelActive}
                contentHeight={contentHeight}
                disabled={disabled}
                togglePanel={togglePanel(index)}
              />
            </div>
          );
        })}
      </div>
      {isExtendable && activePanel === -1 && (
        <Arrow onClick={toggleOpen} open={!open} section="accordion" />
      )}
    </AccordionWrapper>
  );
};

export default Accordion;
