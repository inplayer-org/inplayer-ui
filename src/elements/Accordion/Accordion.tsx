import React, { useState, MouseEvent } from 'react';

// Components
import { Props as TooltipProps } from 'components/Tooltip';
import { AccordionWrapper } from './styled';
import AccordionPanel from './AccordionPanel';
import Arrow from '../../components/NavBar/Arrow';

// Interfaces
interface Panel {
  label: string;
  icon?: Node;
  iconTooltip?: TooltipProps;
  renderContent: () => any;
  disabled: boolean;
}

interface Props {
  panels: Array<Panel>;
  contentHeight: string;
  width?: string;
  extendWidth?: string;
  isExtendable?: boolean;
  onActivePanelChange?: (index: number) => void;
}

const Accordion: React.FC<Props> = ({
  panels,
  contentHeight,
  width,
  extendWidth,
  isExtendable,
  onActivePanelChange,
}: Props) => {
  const [activePanel, setActivePanel] = useState(0);
  const [open, setOpen] = useState(false);

  const openPanel = (panelIndex: number) => {
    if (panelIndex !== activePanel) {
      setActivePanel(panelIndex);
      if (onActivePanelChange) onActivePanelChange(panelIndex);
    }
  };

  const closePanel = (e?: MouseEvent<HTMLButtonElement>) => {
    if (e) e.stopPropagation();
    setActivePanel(-1);
  };

  const togglePanel = (panelIndex: number) => (e: MouseEvent<HTMLButtonElement>) => {
    e.stopPropagation();
    if (activePanel !== -1) {
      closePanel();
    } else {
      openPanel(panelIndex);
    }
  };

  const toggleOpen = () => {
    if (isExtendable) setOpen(open);
  };

  return (
    <AccordionWrapper
      width={width}
      extendWidth={extendWidth}
      open={open}
      contentHeight={contentHeight}
    >
      <div>
        {panels.map((panel, index) => {
          const { icon, iconTooltip, label, renderContent, disabled } = panel;
          const isActive = activePanel === index;
          const isOtherPanelActive = !isActive && activePanel !== -1;

          return (
            <div key={index}>
              <AccordionPanel
                key={index}
                isActive={isActive}
                icon={icon}
                iconTooltip={iconTooltip}
                label={label}
                renderContent={renderContent}
                closePanel={closePanel}
                isOtherPanelActive={isOtherPanelActive}
                contentHeight={contentHeight}
                disabled={disabled}
                togglePanel={togglePanel(index)}
              />
            </div>
          );
        })}
      </div>
      {isExtendable && activePanel === -1 && (
        <Arrow onClick={toggleOpen} open={open} section="accordion" />
      )}
    </AccordionWrapper>
  );
};

Accordion.defaultProps = {
  width: '100%',
  extendWidth: '20%',
  isExtendable: false,
  onActivePanelChange: undefined,
};

export default Accordion;
