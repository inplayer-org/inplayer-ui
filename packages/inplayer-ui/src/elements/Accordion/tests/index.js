import React from 'packages/inplayer-ui/src/elements/Accordion/tests/react';
import { generateSnapshots } from 'packages/inplayer-ui/src/elements/Accordion/tests/helpers';
import Accordion from '../index';

const actionButton = () => <button type="button">Save</button>;

generateSnapshots(Accordion, {
  panels: [
    {
      label: 'Accordion',
      icon: '',
      content: 'Lorem Ipsum',
      renderActionButton: () => actionButton,
      renderFooterLink: () => actionButton,
      // eslint-disable-next-line react/display-name
      renderContent: () => <div>Lorem Ipsum</div>,
    },
  ],
});
