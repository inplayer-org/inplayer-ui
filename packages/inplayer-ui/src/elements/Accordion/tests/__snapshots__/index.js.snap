// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion renders correctly 1`] = `
.c0 {
  width: 100%;
  margin-left: 0%;
  position: relative;
  background: #fff;
  border-left: 1px solid #d7dde5;
  box-sizing: border-box;
  border-top: 1px solid #d7dde5;
  overflow-y: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-transition: width 0.3s ease;
  transition: width 0.3s ease;
}

.c3 {
  font-family: Roboto;
  font-weight: 100;
  color: #286090;
  font-size: 1.25rem;
}

.c5 {
  width: 100%;
  height: 3.5rem;
  box-sizing: border-box;
  overflow: hidden;
  z-index: 10;
  background: #fff;
  -webkit-transition: ease 500ms height;
  transition: ease 500ms height;
  display: none;
}

.c1 {
  margin: 0;
  padding: 1rem 3%;
  width: 100%;
  box-sizing: border-box;
  background: #fff;
  cursor: pointer;
  border-bottom: 1px solid #F6F8FB;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  font-weight: 100;
  color: #3e4348;
  margin: 0;
  display: inline;
}

.c4 {
  display: inline-block;
  font-size: 1rem;
  color: #07AAE6;
}

.c6 {
  padding: 2rem 6%;
}

.c7 {
  position: absolute;
  bottom: -80px;
  width: 100%;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.8rem 3%;
  background: #f0f3f6;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition-timing-function: ease,ease;
  transition-timing-function: ease,ease;
  -webkit-transition-duration: 400ms,1000ms;
  transition-duration: 400ms,1000ms;
  -webkit-transition-property: all,background;
  transition-property: all,background;
  -webkit-transition-delay: 0s,300ms;
  transition-delay: 0s,300ms;
  z-index: 10;
  border-top: 1px solid #d7dde5;
  border-bottom: 1px solid #d7dde5;
}

<div
  className="c0"
  open={false}
  width="100%"
>
  <div>
    <div>
      <header
        className="c1"
        onClick={[Function]}
      >
        <h6
          className="c2 c3"
          color=""
          style={Object {}}
        >
          Accordion
        </h6>
        <span
          aria-hidden={true}
          className="fa fa- c4"
        />
      </header>
      <div
        className="c5"
      >
        <div
          className="c6"
        />
      </div>
      <footer
        className="c7"
      />
    </div>
  </div>
</div>
`;
